numpy/ma/core.py

```diff
@@ -2968,8 +2968,8 @@ def __array_wrap__(self, obj, context=None):
                 np.copyto(result, fill_value, where=d)
                 # Update the mask
                 if m is nomask:
-                    if d is not nomask:
-                        m = d
+                    if any(d):
+                        m = True if all(d) else d
```

Comment:
Is this part necessary? Or is it OK to set the mask to a full array of `True`? It is not tested. (also, write as `d.all()`?)